src/shared/bootspec.c

```diff
@@ -793,7 +786,17 @@ static int find_cmdline_section(
                 *ret_cmdline = NULL;
         else if (r < 0)
                 return log_warning_errno(r, "Failed to read .cmdline section of '%s': %m", path);
-        else if (ret_cmdline)
+
+        word = strdup(cmdline);
+        if (!word)
+                return log_oom();
+
+        /* Quick test to check if there is actual content in the addon cmdline */
+        t = delete_chars(word, NULL);
+        if (t[0] == 0)
+                *ret_cmdline = NULL;
+
+        if (ret_cmdline)
```

Comment:
You'll want an `else` here instead - `ret_cmdline` is already checked above so there's no need to check ti again, and you don't want to overwrite it if the check above is true